### PR TITLE
Adjust mobile navbar

### PR DESCRIPTION
**Pull request by aureentuluvacal**
Addresses #492 

Reformatted mobile navbar, removes the intermediate tab "page" in favor of directly rendering the section.

image

image

### CODE REVIEW

File: app/routes/poll-network.tsx [hunk 1]
```diff
@@ -16,9 +16,14 @@ export async function loader(): Promise<LoaderData> {
   }
 }
 
+const LS_KEY = "flow_network_data"
+
 export default function () {
   const networkData = useLoaderData<LoaderData>()
-  const [status, setStatus] = useState(networkData)
+  const cachedStatus = localStorage.getItem(LS_KEY)
```
Comment (joekrill) [inline app/routes/poll-network.tsx:23]: This won't work if rendered server-side, will it? I don't think `localStorage` will be available on a server render. https://remix.run/docs/en/v1/guides/constraints#rendering-with-browser-only-apis

File: app/routes/poll-network.tsx [hunk 2]
```diff
@@ -40,28 +45,47 @@ export default function () {
   useEffect(() => {
     if (fetcher.data) {
       setStatus(fetcher.data)
+      localStorage.setItem(LS_KEY, JSON.stringify(status))
```
Comment (joekrill) [inline app/routes/poll-network.tsx:48]: I think this is going to cause an outdated version (the previous value) of status in `localStorage`, won't it? Should this be `JSON.stringify(fetcher.data)`?

File: app/ui/design-system/src/lib/Components/NavigationBar/MobileMenu.tsx [hunk 3]
```diff
@@ -1,39 +1,54 @@
...
 
 export type MobileMenuProps = {
   menuItems: MenuItem[]
 }
 
 export function MobileMenu({ menuItems }: MobileMenuProps) {
-  const [selectedIndex, setSelectedIndex] = useState(-1)
-  const selectedMenuItem = menuItems[selectedIndex] as
-    | DropdownMenuItem
+  const [selectedIndex, setSelectedIndex] = useState<[number, number]>([-1, -1])
```
Comment (joekrill) [inline app/ui/design-system/src/lib/Components/NavigationBar/MobileMenu.tsx:12]: This is a little confusing to me. Why do we need two indexes and what is the difference between them? Can we make this more clear maybe?

**Review by joekrill — CHANGES_REQUESTED** [review]

File: app/routes/poll-network.tsx (hunk 1 above)
Comment (aureentuluvacal) [reply app/routes/poll-network.tsx:23]: I'll just take this out for now. Keep forgetting remix does SSR.

**Pull request merged**